components/external-services-mock/internal/selfreg/handler.go

```diff
@@ -70,15 +75,36 @@ func (h *Handler) HandleSelfRegPrep(w http.ResponseWriter, r *http.Request) {
 		httphelpers.WriteError(w, err, http.StatusBadRequest)
 		return
 	}
+
+	SelfRegistrations[name] = tenant
 }
 
 func (h *Handler) HandleSelfRegCleanup(w http.ResponseWriter, r *http.Request) {
 	params := mux.Vars(r)
-	if name, ok := params[NamePath]; !ok || name == "" {
+	name, ok := params[NamePath]
+	if !ok || name == "" {
 		err := errors.New(fmt.Sprintf("%s is missing from path", NamePath))
 		httphelpers.WriteError(w, err, http.StatusBadRequest)
 		return
 	}
+
+	providerSubaccount, cloningExists := SelfRegistrations[name]
```

Comment:
We need to be sure that external services return error in that case, since it is pretty standard to return 200 OK if something that does not exists is deleted.

Reply:
About the error, yes. We could leave it like it is because personally, I'm not sure if the external service returns an error. But we could add the non-empty check for the `providerSubacccount`